internal/service/batch/job_definition.go

```diff
@@ -166,12 +184,21 @@ func ResourceJobDefinition() *schema.Resource {
 					},
 				},
 			},
+
 			"revision": {
 				Type:     schema.TypeInt,
 				Computed: true,
 			},
+
+			"scheduling_priority": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				ValidateFunc: validation.IntBetween(0, 9999),
```

Comment:
I usually try to avoid these validate funcs as they are add to maintaining more logic on ourside, eventually the API would return this error anyways. But I think it's fine to have in our provider too, it's just a personal preference. We should at some point have more clarity on what "logic rules" we want to cover and which not.